src/object/srv_obj.c

```diff
@@ -1981,3 +1990,64 @@ ds_obj_abt_pool_choose_cb(crt_rpc_t *rpc, ABT_pool *pools)
 
 	return pool;
 }
+
+static bool
+cont_prop_srv_verify(struct ds_iv_ns *ns, uuid_t co_hdl)
+{
+	int			rc;
+	daos_prop_t		cont_prop = {0};
+	struct daos_prop_entry	entry = {0};
+
+	entry.dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+	cont_prop.dpp_entries = &entry;
+	cont_prop.dpp_nr = 1;
+
+	rc = cont_iv_prop_fetch(ns, co_hdl, &cont_prop);
+	if (rc != 0)
+		return false;
+	return daos_cont_prop2serververify(&cont_prop);
+}
+
+static int
+ds_csum_verify_bio(crt_rpc_t *rpc, struct bio_desc *biod,
+		   struct daos_csummer *csummer)
+{
+	struct obj_rw_in	*orw = crt_req_get(rpc);
+	struct ds_pool		*pool;
+	daos_iod_t		*iods = orw->orw_iods.ca_arrays;
+	uint64_t		 iods_nr = orw->orw_iods.ca_count;
+	unsigned int		 i;
+	bool			 is_update;
+	int			 rc = 0;
+
+	if (!daos_csummer_initialized(csummer))
+		return 0;
+
+	pool = ds_pool_lookup(orw->orw_pool_uuid);
+	is_update =	opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_UPDATE ||
+			opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_TGT_UPDATE;
+	if (!is_update ||
+	    !cont_prop_srv_verify(pool->sp_iv_ns, orw->orw_co_hdl))
+		return 0;
+
+	for (i = 0; i < iods_nr && rc == 0; i++) {
+		daos_iod_t		*iod = &iods[i];
+		struct bio_sglist	*bsgl  = bio_iod_sgl(biod, i);
+		d_sg_list_t		 sgl;
+		/** Currently only supporting array types */
+		bool			 type_is_supported =
+						iod->iod_type == DAOS_IOD_ARRAY;
```

Comment:
So not support DAOS_IOD_SINGLE yet, right?

Reply:
Correct. Will support it in a later change.